.scripts/automation_init.sh

```diff
@@ -1,4 +1,4 @@
 #!/usr/bin/env bash
-npm install -g @azure-tools/typespec-client-generator-cli@0.14.1
+npm install -g @azure-tools/typespec-client-generator-cli
```

Comment:
@raych1 could you recomment a new version?

Reply:
Use 0.21.0. You might refer to https://github.com/Azure/azure-rest-api-specs/blob/4ba642bf36e93f15819c3c414e6a1c449f1979d6/package.json#L12 in the future.

Reply:
I think just remove @0.14.1, it will install the latest version?

Reply:
why the new automation using a different version than the init script specified 
![image](https://github.com/user-attachments/assets/f214148f-76a7-41b4-bf7f-dfffbf06edca)

Reply:
@kazrael2119 please create an issue from qiaoqiao's comment, I'm going to resolve this conversation to merge, since it's a critial issue to js automation

Reply:
@qiaozha theatrically generation script should use the installed version by the 'init script` under the SDK repo path. Did the installation of `0.14.1` version fail? @kazrael2119 can you check if this can be repoed locally?

Reply:
@raych1 checked, the init script succeeded https://dev.azure.com/azure-sdk/internal/_build/results?buildId=4813695&view=logs&j=47092895-c7c8-567a-c7a6-8c88f3ec178e&t=d293a8e8-c6af-5091-cdf7-4c66482dd6f2&l=378

Reply:
I saw in [the log](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=4813695&view=logs&j=47092895-c7c8-567a-c7a6-8c88f3ec178e&t=d293a8e8-c6af-5091-cdf7-4c66482dd6f2&l=501) `0.14.1` was used in the generation step which is expected.